httpx/client.py

```diff
@@ -332,7 +331,6 @@ class AsyncClient(BaseClient):
         cert: CertTypes = None,
         verify: VerifyTypes = None,
         timeout: TimeoutTypes = None,
-        trust_env: bool = True,
```

Comment:
We don't want to get rid of `trust_env` here - we just want to change the behavior so that if `trust_env == None` then the client default is used, but if `trust_env in (True, False)` then the explicitly requested behaviour is used.